### DISCUSSION TITLE
Replace current ProxyKit code with ReverseProxy

### DISCUSSION THREAD

**Discussion by npalla-tyler** [dotnet/yarp#736] [Q&A] [answered]
Hi All,

Here is my requirement. We replaced our IIS URL rewriting with proxy-kit and everything is OK. Since proxy-kit is obsolete trying to migrate to ReverseProxy as per the Proxy-Kit update. We had an external api and that comes with /otherapi and if the route matches that I need to get data from otherapi to UI.

Proxy-Kit code:
```c#
                var apiUrl = configuration["other:otherapiUrl"];
                if (!apiUrl.EndsWith("/api", StringComparison.InvariantCultureIgnoreCase))
                {
                    apiUrl = apiUrl.TrimEnd('/') + "/api";
                }
                apiUrl = apiUrl.TrimEnd('/');

                // sometimes we have two slashes in the URL coming from the apps.
                builder.Map(@"//otherapi", proxy => { SetupForwardProxy(proxy, apiUrl); });
                builder.Map(@"/otherapi", proxy => { SetupForwardProxy(proxy, apiUrl); });


proxy.RunProxy(context =>
            {
                var upstreamHost = new UpstreamHost(apiUrl);
                // if debugging DebugUri(upstreamHost, context);
                return context
                    .ForwardTo(upstreamHost)
                    .AddXForwardedHeaders()
                    .AddOtherApiHeader()
                    .Send();
            });
```

AddOtherApiHeader - adds a bunch of headers to the UpstreamRequest required for the other api.

I am trying to use Code Sample from the reverse-proxy repo, but could not see a way to dynamically pass custom headers. I see the RouteConfig has the option to add Transform where we can pass during startup. But need to update the header values based on the individual user that logged in. Trying to see if there is any option with respect to the code sample example from the repo. Any reference examples would be greatly appreciated. We can get ClusterConfig and DestinationInfo but here, but not sure how I can add custom headers.
![image](https://user-images.githubusercontent.com/78396632/107685977-a2e63700-6c72-11eb-9249-c7ed8d02f6c1.png)

Requirement is - if api request has **otheapi** in the URL (or starts with it), then I need to get the data using different url by passing bunch of custom headers to the request.

Package version - 1.0.0-preview.9.21109.2

Thanks a lot.

**Comment by Tratcher** [comment, marked as answer]
The proxykit equivalent API is called IHttpProxy: https://microsoft.github.io/reverse-proxy/articles/direct-proxying.html#update-startup
You can manipulate the outgoing requests using the HttpTransformer parameter.
https://github.com/microsoft/reverse-proxy/blob/f4ebaaf60c7194235a73b6131842011e0670bb78/samples/ReverseProxy.Direct.Sample/Startup.cs#L43-L53

**Reply by npalla** [reply]
Thanks @Tratcher!! It working fine but I have issue with double slash.  http://dev.localdev.tcpci.com/app/myapp/extapi/system/settings works fine but http://dev.localdev.tcpci.com/app/myapp//extapi/system/settings  (with double slash before extapi) is throwing 404 (linux container).
Here is the route pattern I am using.
endpoints.Map(@"extapi/{controller}/{action}", async httpContext =>

Any help on this would be great.

Thank you.
